src/ghostfragment/fragmenting/inter_finder.cpp

```diff
@@ -63,26 +89,33 @@ const log_type find_pair_intersections(
     return log;
 }
 
-
-void create_system(const mol_type& mol, system_type& fragments) {
+system_type create_system(const mol_type& mol) {
+    system_type fragments;
     const nuke_type f_nuclei = mol.fragmented_nuclei();
     for(std::size_t i = 0; i < f_nuclei.size(); ++i) {
         auto nukes = f_nuclei[i];
         frag_type fragment;
         for(const auto atom_i : nukes) { fragment.push_back(atom_i); }
         fragments.push_back(fragment);
     }
+    return fragments;
 }
 
 res_type ghostfragment::fragmenting::intersections(const mol_type& mol) {
     res_type m;
-    system_type fragments;
-    create_system(mol, fragments);
-    log_type log = find_pair_intersections(fragments);
+    system_type fragments = create_system(mol);
+    const log_type log    = find_pair_intersections(fragments);
     for(std::size_t i = 0; i < fragments.size(); ++i) {
         inter_type inter(fragments[i]);
         group_type group({i});
-        find_group_intersections(fragments, inter, i, m, log, group);
+        res_type new_m =
+          find_group_intersections(fragments, inter, i, m, log, group);
```

Comment:
I think you want to do something like:
```suggestion
        m =
          find_group_intersections(fragments, inter, i, std::move(m), log, group);
```
basically you pass `m` into `find_group_intersections`, `find_group_intersections` updates it, and the returns the updated version of `m`. This is more or less equivalent to what you had when you were passing by reference, just cleaner.